File: backend/app/schemas/item.py
from pydantic import BaseModel
from typing import Dict, Any, Optional


class ItemBase(BaseModel):
    name: str
    type: str
    description: Optional[str] = None
    effect: Dict[str, Any]
    price: int
    rarity: str


class Item(ItemBase):
    id: str

    class Config:
        from_attributes = True


class ItemResponse(Item):
    pass

File: backend-java/src/main/java/com/aetherisland/entity/Item.java
package com.aetherisland.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.util.Map;

@Entity
@Table(name = "items")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Item {
    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    private String id;
    
    @Column(nullable = false)
    private String name;
    
    @Column(nullable = false)
    private String type;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @Column(columnDefinition = "jsonb")
    @JdbcTypeCode(SqlTypes.JSON)
    private Map<String, Object> effect;
    
    @Column(nullable = false)
    private Integer price = 0;
    
    @Column(nullable = false)
    private String rarity = "common";
}

